### ISSUE TITLE
A11Y: Design > Layers

### ISSUE THREAD

**Issue by dreamofabear** [GoogleForCreators/web-stories-wp#4432]
## Solution to be implemented

1. Hide entire layer panel for screen readers by using `aria-hidden="true"`
1. Display layer manipulation shortcuts in the keyboard shortcut dialog:
  a. <kbd>⌘</kbd><kbd>↑</kbd> - move element forward
  b. <kbd>⌘</kbd><kbd>↓</kbd> - move element backward
  c. <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↑</kbd> - move element to front
  d. <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↓</kbd> - move element to back
#### 1. Navigating into Layers section blocked after expansion.

- User enables VoiceOver while Layers is collapsed. 
- User expands Layers via the arrow toggle button. 
- Expected: User selects "next item" in screen reader and the layers list group is selected.
- Actual: User is unable to navigate beyond the toggle button into the list group of layers via "next item" VoiceOver navigation. 
  - User's VoiceOver navigation actually becomes trapped between the "Layers" label and the toggle button.

#### 2. Layer items in list group are announced twice.

- User enables VoiceOver while Layers is expanded.
- User navigates into the Layers list group.
- Expected: Each layer is announced once with its layer contents (e.g. text, image, video).
- Actual: Each layer is preceded by a blank group (e.g. "group 1 of 16" with 8 layers).

## Additional Context

- Plugin Version: 1.0.0-beta.2+fd68af7
- Operating System: macOS 10.15.6
- Browser: Chrome 85

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by bmattb** [comment]
@o-fernandez can you confirm that this should be prioritized by Prometheus for Stable in Sprint 37 right now

**Comment by dreamofabear** [comment]
Re: (1), "keyboard focus trapped" can be worked around with the new keyboard shortcuts (#4454). Command+Shift+1/2/3 (macOS) switches focus between element/workspace/design panels.

**Comment by dreamofabear** [comment]
We've received some pushback on the workaround for (1). 

@barklund WDYT about triggering a keyboard focus change to the Layer list-group after expand button is clicked?

**Comment by barklund** [comment]
> > Re: (1), "keyboard focus trapped" can be worked around with the new keyboard shortcuts (#4454). Command+Shift+1/2/3 (macOS) switches focus between element/workspace/design panels.
> We've received some pushback on the workaround for (1).

Oh, my point was actually to not use the layer panel at all when using a keyboard (not even focusing it), but using keyboard shortcuts on the elements themselves directly:

* <kbd>⌘</kbd><kbd>↑</kbd> - move currently selected element one layer forward
* <kbd>⌘</kbd><kbd>↓</kbd> - move currently selected element one layer backward
* <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↑</kbd> - move currently selected element top-most
* <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↓</kbd> - move currently selected element bottom-most

You are of course not able to simply "read" the current layer ordering this way, but you are able to manipulate the layers using keyboard-only.

Is this sufficient, or do you need to be able to "read" the layer order as well?

**Comment by dreamofabear** [comment]
As discussed offline, let's do two things for now to mostly resolve this issue:
- [ ] Add these shortcuts to the in-app keyboard shortcut guide
- [ ] Hide the Layers panel from the accessibility tree

**Comment by csossi** [comment]
Verified in QA
